noir_stdlib/src/sha256.nr

```diff
@@ -16,8 +16,8 @@ fn msg_u8_to_u32(msg: [u8; 64]) -> [u32; 16] {
     msg32
 }
 // SHA-256 hash function
-pub fn digest<N>(msg: [u8; N]) -> [u8; 32] {
-    sha256_var(msg, N)
+pub fn digest<let N: u32>(msg: [u8; N]) -> [u8; 32] {
+    sha256_var(msg, N as u64)
```

Comment:
I figured we could change the signature of `sha256_var` to just be a `u32`. Even though it is a minor change it is still a breaking change though so I decided to leave it as `u64` for the time being as to keep this PR non-breaking.